lib/ecto/migrator.ex

```diff
@@ -673,12 +675,36 @@ defmodule Ecto.Migrator do
     |> Enum.sort()
   end
 
+  # See: https://github.com/elixir-ecto/ecto_sql/issues/599
+  defp warn_for_misnamed_files(directory) do
+    [directory, "**", "*.ex"]
+    |> Path.join()
+    |> Path.wildcard()
+    |> Enum.each(fn path ->
+      if Path.basename(path) =~ ~r/\d+_/ do
+      end
+    end)
+  end
+
   defp extract_migration_info(file) do
     base = Path.basename(file)
 
     case Integer.parse(Path.rootname(base)) do
-      {integer, "_" <> name} -> {integer, name, file}
-      _ -> nil
+      {integer, "_" <> name} ->
+        if Path.extname(base) == ".ex" do
+          IO.warn("""
+          File #{Path.relative_to_cwd(file)} looks like a migration but ends in .ex. \
+          Migration files should end in .exs. Use "mix ecto.gen.migration" to generate \
+          migration files with the correct extension.\
+          """)
```

Comment:
```suggestion
          IO.warn("""
          file #{Path.relative_to_cwd(file)} looks like a migration but ends in .ex. \
          Migration files should end in .exs. Use "mix ecto.gen.migration" to generate \
          migration files with the correct extension\
          """, [])
```

I added `[]` at the end because the stacktrace comes from Ecto guts and that won't be helpful? Another option is to pass `file: file` as the argument to IO.warn and that will be included in the warning message, so you don't need it in the message itself.